examples/test_api.c

```diff
@@ -185,7 +190,59 @@ int test_abtoull(char* str, unsigned long long* val)
   return SCR_SUCCESS;
 }
 
-double getbw(char* name, char* buf, size_t size, int times)
+size_t get_my_file_offset()
+{
+  uint64_t file_offset = 0;
+
+  if (use_shared_file) {
+    uint64_t send_buf = my_filesize;
+    MPI_Scan(&send_buf, &file_offset, 1, MPI_UINT64_T, MPI_SUM, MPI_COMM_WORLD);
+    file_offset -= my_filesize;
+  }
+
+  return file_offset;
+}
+
+int create_file(char* file)
+{
+  int fd;
+
+  if (use_shared_file) {
+    if (rank == 0) {
+      if ( (fd = open(file, O_WRONLY | O_CREAT | O_TRUNC, S_IRUSR | S_IWUSR)) > 0) {
+        if (truncate(file, total_filesize) < 0) {
+          printf("%d: Could not truncate file %s : %s\n", rank, file, strerror(errno));
+          close(fd);
+          fd = -1;
+        }
+      }
+      else {
+        printf("%d: Could not create file %s : %s\n", rank, file, strerror(errno));
+      }
+
+      if (fd > 0) {
+        close(fd);
+      };
+    }
+
+    MPI_Bcast(&fd, 1, MPI_INT, 0, MPI_COMM_WORLD); /* Wait for rank 0 to complete creation */
+
+    if (fd > 0) {
```

Comment:
oh, wait.  I think ``fd==0`` also means success, right?

Reply:
and if that's true, I have that bug all throughout the test_*.c code.

Reply:
whoops!

Reply:
Yeah, I had noticed that too.  But it is rare that stdin, stdout, and/or stderr would be closed.  But I suppose it could happen.  I'll clean this up